clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -495,6 +540,10 @@ private void updateGroupSubscription(Set<String> topics) {
 
         Map<String, Assignment> assignments = assignor.assign(metadata.fetch(), new GroupSubscription(subscriptions)).groupAssignment();
 
+        if (protocol == RebalanceProtocol.COOPERATIVE) {
+            validateCooperativeAssignment(ownedPartitions, assignments);
```

Comment:
How critical do you think this validation is? Since there are so few assignors in the wild, I think I'm wondering if we can just document the requirement and leave it up to the assignor. We don't do other validations such as verifying disjoint assignments.

Reply:
My arguments come from this: there are different levels of severity from ill-behaved assignor:

1) duplicate consumption: one partition being consumed by multiple members, this can also be caused by non-disjoint assignments.

2) runtime error: the consumer would just break and die; this can be caused by incorrect movement, e.g. `consumer.position` returns an IllegalStateException.

I think it is okay to falls into 1) due to ill-behaved assignor, but 2) should better be avoided, and to do so it is reasonable to fail-early.

Reply:
I don't feel too strongly one way or another, but it seems to me like 1) is more similar to what we're avoiding by doing this validation, ie guarding against a partition being assigned when it has not yet been fully revoked and is therefore still (sort of ) assigned to two members?
I do agree it seems a bit weird to validate this but accept possibly overlapping assignments from poorly behaving assignor implementations in other cases

Reply:
I don't feel too strongly. I think it may be useful initially for users who are trying to figure out how cooperative assignment is supposed to work. Its semantics are more subtle than eager assignment. I'm not sure I understand how case 2) could happen though even with a buggy assignor.